frontend/src/lib/URLParser.test.ts

```diff
@@ -66,9 +69,9 @@ describe('URLParser', () => {
   it('does not create new state when setting query param in-place', () => {
     location.search = '?searchkey=searchvalue';
 
-    const historyLength = history.length;
+    const historyLength = window.history.length;
```

Comment:
The `length` property of `history` is [no longer available](https://github.com/remix-run/history/blob/main/docs/api-reference.md), so I replaced it here with `window.history`. Since `window.history` does not have the `search` property, I did not replace the full `history` object, just the instances that used `length`.